adafruit_requests.py

```diff
@@ -529,6 +529,18 @@ def _get_socket(
         self._socket_free[sock] = False
         return sock
 
+    @staticmethod
+    def _check_headers(headers: Dict[str, str]):
+        if not isinstance(headers, dict):
+            raise AttributeError("headers must be in dict format")
+
+        for key, value in headers.items():
+            if isinstance(value, (str, bytes)) or value is None:
+                continue
+            raise AttributeError(
+                f"Header part ({value}) from {key} must be of type str or bytes, not {type(value)}"
+            )
```

Comment:
in `CPython requests` [this is](https://github.com/psf/requests/blob/96b22fa18c00831656ee4b286bf1c9062459b00a/src/requests/utils.py#L1040):
```
        raise InvalidHeader(
            f"Invalid leading whitespace, reserved character(s), or return "
            f"character(s) in header {header_kind}: {header_part!r}"
        )
```
An `AttributeError` seems similar enough and didn't know if creating a new Exception would be wanted